### DISCUSSION TITLE
ScriptReference/MaterialPropertyDrawer

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#312] [Notes]
# ScriptReference/MaterialPropertyDrawer

https://docs.unity3d.com/ScriptReference/MaterialPropertyDrawer.html

**Comment by SoylentGraham** [comment]
You can add custom parameters to your custom drawer by adding a matching constructor.  
`[RangeMinMax(0,100)]SomeValue("SomeValue", VECTOR ) = (0,0,0,0)`  
and in C#  

```csharp
public class RangeMinMaxDrawer : MaterialPropertyDrawer  
{  
 float Min,Max;  
 public RangeMinMaxDrawer(float Min, float Max)  
 {  
  this.Min = Min;  
  this.Max = Max;  
  }  
}
```

**Comment by SoylentGraham** [comment]
One option not listed here, but built-in, is  
`[HideInInspector]`  
To hide (but still reflect)
